### PR TITLE
Refresh plugin

### PR DESCRIPTION
**Pull request by basil**
CC @kuisathaverat

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -25,7 +26,8 @@ under the License.
   <parent>
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>4.49</version>
+    <version>4.50</version>
+    <relativePath />
```
Comment (basil) [inline pom.xml:30]: Consistent with the archetype.

File: pom.xml [hunk 2]
```diff
@@ -35,7 +37,7 @@ under the License.
   <url>https://github.com/jenkinsci/${project.artifactId}-plugin</url>
 
   <scm>
-    <connection>scm:git:git://github.com/${gitHubRepo}.git</connection>
+    <connection>scm:git:https://github.com/${gitHubRepo}.git</connection>
```
Comment (basil) [inline pom.xml:40]: GitHub has deprecated `git://` URLs.

File: pom.xml [hunk 3]
```diff
@@ -219,30 +221,18 @@ under the License.
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>jackson2-api</artifactId>
-      <version>2.13.4.20221013-295.v8e29ea_354141</version>
```
Comment (basil) [inline pom.xml:222]: Managed in plugin BOM.

File: pom.xml [hunk 4]
```diff
@@ -219,30 +221,18 @@ under the License.
...
       <groupId>org.mockito</groupId>
       <artifactId>mockito-core</artifactId>
       <scope>test</scope>
     </dependency>
-    <dependency>
-      <groupId>org.hamcrest</groupId>
-      <artifactId>java-hamcrest</artifactId>
-      <version>2.0.0.0</version>
-      <scope>test</scope>
-    </dependency>
```
Comment (basil) [inline pom.xml:234]: Managed in plugin parent POM.

File: pom.xml [hunk 5]
```diff
@@ -219,30 +221,18 @@ under the License.
...
     <dependency>
       <groupId>io.jenkins</groupId>
       <artifactId>configuration-as-code</artifactId>
       <scope>test</scope>
     </dependency>
-    <dependency>
-      <groupId>junit</groupId>
-      <artifactId>junit</artifactId>
-      <scope>test</scope>
-    </dependency>
```
Comment (basil) [inline pom.xml:245]: Managed in plugin parent POM.

File: pom.xml [hunk 6]
```diff
@@ -274,10 +264,6 @@ under the License.
         <type>pom</type>
         <scope>import</scope>
       </dependency>
-      <dependency>
-        <groupId>org.cryptacular</groupId>
-        <artifactId>cryptacular</artifactId>
-      </dependency>
```
Comment (basil) [inline pom.xml:280]: No version number, so dead code.

File: src/main/java/org/jenkinsci/plugins/saml/SamlFormValidation.java [hunk 7]
```diff
@@ -58,9 +58,6 @@ public static FormValidation checkEmailFormat(String value, String message) {
         catch (AddressException ae) {
             return FormValidation.error(message);
         }
-        catch (NullPointerException ne) {
-            return FormValidation.warning(message);
-        }
```
Comment (basil) [inline src/main/java/org/jenkinsci/plugins/saml/SamlFormValidation.java:63]: This whole method appears to be dead code, but it trips up new versions of SpotBugs. Anyway it does not appear that a `NullPointerException` could actually occur here even if this method was invoked.

**Review by kuisathaverat — APPROVED** [review]

**Pull request merged**